pkg/controller/migstorage/validation.go

```diff
@@ -168,73 +161,79 @@ func (r ReconcileMigStorage) validateVolumeSnapshotStorage(storage *migapi.MigSt
 	// Secret
 	secret, err := storage.GetVolumeSnapshotCredSecret(r)
 	if err != nil {
-		return liberr.Wrap(err)
+		return liberr.Wrap(fmt.Errorf("error getting migstorage backup provider secret %s/%s: %v",
+			storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Namespace,
+			storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Name,
+			err))
 	}
 
 	if storage.Spec.VolumeSnapshotProvider != "" {
 		// Unknown provider.
 		if provider == nil {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSProvider,
 				Status:   True,
 				Reason:   NotSupported,
 				Category: Critical,
-				Message:  InvalidVSProviderMessage,
-			})
+				Message:  fmt.Sprintf("could not get provider for spec.VolumeSnapshotProvider: %s", storage.Spec.VolumeSnapshotProvider),
+			}, storage.Namespace, storage.Name)
 			return nil
 		}
 
 		// NotSet
 		if !migref.RefSet(settings.CredsSecretRef) {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSCredsSecretRef,
 				Status:   True,
 				Reason:   NotSet,
 				Category: Critical,
-				Message:  InvalidVSCredsSecretRefMessage,
-			})
+				Message:  "spec.VolumeSnapshotConfig.CredsSecretRef name or namespace is empty",
+			}, storage.Namespace, storage.Name)
 			return nil
 		}
 
 		// NotFound
 		if secret == nil {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSCredsSecretRef,
 				Status:   True,
 				Reason:   NotFound,
 				Category: Critical,
-				Message:  InvalidVSCredsSecretRefMessage,
-			})
+				Message: fmt.Sprintf("VolumeSnapshotConfig secret %s/%s not found",
+					storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Namespace,
+					storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Namespace),
+			}, storage.Namespace, storage.Name)
 			return nil
 		}
 
 		// Fields
 		fields := provider.Validate(secret)
 		if len(fields) > 0 {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSFields,
 				Status:   True,
 				Reason:   NotSupported,
 				Category: Critical,
-				Message:  InvalidVSFieldsMessage,
```

Comment:
I see that all of these const messages are no longer being used. Should we prune the consts that are no longer needed?